includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -2111,13 +2111,15 @@ protected function get_primary_key_for_lookup_table( $table ) {
 	 */
 	public function get_query_for_stock( $product_id ) {
 		global $wpdb;
-		return $wpdb->prepare(
+		$query = $wpdb->prepare(
 			"
 			SELECT COALESCE ( MAX( meta_value ), 0 ) FROM $wpdb->postmeta as meta_table
 			WHERE meta_table.meta_key = '_stock'
 			AND meta_table.post_id = %d
 			",
 			$product_id
 		);
+
+		return apply_filters( 'woocommerce_query_for_stock', $query, $product_id );
```

Comment:
The name of this query is likely too broad given how specific its implementation is.